rs/crypto/internal/crypto_service_provider/src/vault/remote_csp_vault/tarpc_csp_vault_client.rs

```diff
@@ -218,11 +236,18 @@ impl RemoteCspVaultBuilder {
             TarpcCspVaultClient::new(Default::default(), transport).spawn()
         };
         debug!(self.logger, "Instantiated remote CSP vault client");
+        let internal_rt_for_safe_block_on = tokio::runtime::Builder::new_multi_thread()
+            .worker_threads(1)
```

Comment:
Could be good to add a comment here for why multi_thread runtime is needed. I assume this quote from the blog you posted explains it?

>In this example, it is important that the runtime is configured to be a [multi_thread](https://docs.rs/tokio/1/tokio/runtime/struct.Builder.html#method.new_multi_thread) runtime. If you change it to be a [current_thread](https://docs.rs/tokio/1/tokio/runtime/struct.Builder.html#method.new_current_thread) runtime, you will find that the time consuming task finishes before any of the background tasks start. This is because background tasks spawned on a current_thread runtime will only execute during calls to block_on as the runtime otherwise doesn't have anywhere to run them